corehq/apps/users/views/mobile/users.py

```diff
@@ -679,6 +685,226 @@ def post(self, request, *args, **kwargs):
+                .size(limit))
+
+    @allow_remote_invocation
+    def get_pagination_data(self, in_data):
+        if not isinstance(in_data, dict):
+            return {
+                'success': False,
+                'error': _("Please provide pagination info."),
+            }
+        try:
+            limit = int(in_data.get('limit', 10))
+        except ValueError:
+            limit = 10
+
+        page = in_data.get('page', 1)
+        query = in_data.get('query')
+
+        users_query = self._user_query(query, page, limit)
+        if in_data.get('showDeactivatedUsers', False):
+            users_query = users_query.show_only_inactive()
+        users_data = users_query.run()
+        return {
+            'response': {
+                'itemList': map(self._format_user, users_data.hits),
+                'total': users_data.total,
+                'page': page,
+                'query': query,
+            },
+            'success': True,
+        }
+
+    @allow_remote_invocation
+    def modify_user_status(self, in_data):
+        if not self.can_add_extra_users:
+            return {
+                'error': _("No Permission."),
+            }
+        try:
+            user_id = in_data['user_id']
+        except KeyError:
+            return {
+                'error': _("Please provide a user_id."),
+            }
+        try:
+            is_active = in_data['is_active']
+        except KeyError:
+            return {
+                'error': _("Please provide an is_active status."),
+            }
+        user = CommCareUser.get_by_user_id(user_id, self.domain)
+        user.is_active = is_active
+        user.save()
+        return {
+            'success': True,
+        }
+
+    @allow_remote_invocation
+    def check_username(self, in_data):
+        try:
+            username = in_data['username'].strip()
+        except KeyError:
+            return HttpResponseBadRequest('You must specify a username')
+        if username == 'admin' or username == 'demo_user':
+            return {'error': _('Username {} is reserved.'.format(username))}
+        if '@' in username:
+            return {
+                'error': _('Username {} cannot contain "@".'.format(username))
+            }
+        if ' ' in username:
+            return {
+                'error': _('Username {} cannot contain '
+                           'spaces.'.format(username))
+            }
+        full_username = format_username(username, self.domain)
+        if CommCareUser.get_by_username(full_username, strict=True):
+            result = {'error': _('Username {} is already taken'.format(username))}
+        else:
+            result = {'success': _('Username {} is available'.format(username))}
+        return result
+
+    @allow_remote_invocation
+    def create_mobile_worker(self, in_data):
+        if not self.can_add_extra_users:
+            return {
+                'error': _("No Permission."),
```

Comment:
This could probably be expounded upon.  Maybe "You have reached the limit of users available under your current subscription" or something.